app/models/spree/credit_card_decorator.rb

```diff
@@ -14,6 +14,9 @@
 
   belongs_to :user
 
+  after_create :ensure_default
+  after_save :ensure_default, if: :is_default_changed?
```

Comment:
Is `:is_default_changed?` something AR adds??

Reply:
Yep, the name of the attribute followed by `_changed?` tells you whether it is being updated.
http://api.rubyonrails.org/v3.2.19/classes/ActiveModel/Dirty.html